### PR TITLE
Add search view behind feature flag

### PR DESCRIPTION
**Pull request by richtier**
[this task](https://uktrade.atlassian.net/browse/ED-1273)

The template and style will be improved by @stevenburnell-dit as a follow up task.

image

### CODE REVIEW

File: company/views.py [hunk 1]
```diff
@@ -25,6 +27,36 @@ def get(self, request, *args, **kwargs):
...
+        return super().get_context_data(
+            active_view_name='public-company-profiles-list',
+            **kwargs,
+        )
+
+    def get_results(self, form):
+        response = api_client.company.search(
+            term=form.cleaned_data['term']
+        )
+        if not response.ok:
```
Comment (konradko) [inline company/views.py]: could be just `response.raise_for_status()` as if the response is `200` it returns `None`: http://docs.python-requests.org/en/master/user/quickstart/#response-status-codes

File: docker/env.json [hunk 2]
```diff
@@ -23,7 +23,8 @@
       "GOOGLE_TAG_MANAGER_ENV",
       "FEATURE_ADVANCED_MANUFACTURING_ENABLED",
       "FEATURE_MORE_INDUSTRIES_BUTTON_ENABLED",
-      "FEATURE_UNSUBSCRIBE_VIEW_ENABLED"
+      "FEATURE_UNSUBSCRIBE_VIEW_ENABLED",
```
Comment (konradko) [inline docker/env.json]: that got removed?

File: ui/settings.py [hunk 3]
```diff
@@ -127,6 +127,9 @@
 FEATURE_UNSUBSCRIBE_VIEW_ENABLED = (
     os.getenv('FEATURE_UNSUBSCRIBE_VIEW_ENABLED') == 'true'
 )
+FEATURE_COMPANY_SEARCH_VIEW_ENABLED = (
+    os.getenv('FEATURE_COMPANY_SEARCH_VIEW_ENABLED') == 'true'
```
Comment (konradko) [inline ui/settings.py]: shouldn't the default be `false`, given this is WIP?

File: ui/tests/test_context_processors.py [hunk 4]
```diff
@@ -12,15 +12,16 @@ def test_feature_flags_installed(settings):
 
 def test_feature_returns_expected_features(rf, settings):
 
-    settings.FEATURE_MORE_INDUSTRIES_BUTTON_ENABLED = True
-
+    settings.FEATURE_MORE_INDUSTRIES_BUTTON_ENABLED = 'a'
```
Comment (konradko) [inline ui/tests/test_context_processors.py]: please use the actual values these settings may have :)

File: ui/settings.py (hunk 3 above)
Comment (richtier) [inline ui/settings.py]: I take it you're suggesting we do this:  `os.getenv('THE_FEATURE_FLAG', 'false') == 'true'`?

 `os.getenv('THE_FEATURE_FLAG') == 'true'` evaluating to `False` by default already, and the other WIP feature flags also do not explicitly have `false`, so for consistency  `os.getenv('THE_FEATURE_FLAG') == 'true'` is fine.

File: ui/tests/test_context_processors.py (hunk 4 above)
Comment (richtier) [inline ui/tests/test_context_processors.py]: using `'a'` and '`'b'` detects if the following mistake is made in the context processor:
```
'features': {
    'some_var': settings.some_var,
    'some_other_var': settings.some_var,  # misatake: this should be `settings.some_other_var
}
```

if that happened then the following tests would not detect the mistake:

```
settings.some_var = True
settings.some_other_var = True
```

but the following would:

```
settings.some_var = 'a'
settings.some_other_var = 'b'
```

Having said that, as there are only two values in play, one can be `True`, and other `False`. The problem I'm describing only needs solving wheen there are more than once value.

File: ui/settings.py (hunk 3 above)
Comment (konradko) [inline ui/settings.py]: sorry, was thinking about default

**Pull request merged**